compiler/rustc_const_eval/src/check_consts/resolver.rs

```diff
@@ -248,10 +248,10 @@ where
 #[derive(Debug, PartialEq, Eq)]
 pub(super) struct State {
     /// Describes whether a local contains qualif.
-    pub qualif: BitSet<Local>,
+    pub qualif: MixedBitSet<Local>,
```

Comment:
Might be worth a comment explaining the choice of representation?

Reply:
I can do that. In general using a `BitSet` on large domain sizes can create issues like this.

Reply:
There's no way a contributor could know this. `BitSet` is the most canonically named type so it's the one developers will reach for by default.

Reply:
I don't disagree. Thinking about the domain representation and whether it's dense or sparse is super common in dataflow so "no way to know this" is probably too broad of a brush stroke though. 

We could add a comment to `BitSet`, or the `bit_set` module, describing and linking to the other bitsets, cc @nnethercote on that.

In this case I've added a comment, if you're happy with it feel free, to r you and michael.

Reply:
I think it would be worth renaming BitSet to DenseBitSet perhaps? That way there's not an obvious one to reach for.

But that should be follow-up for sure :)

Reply:
Sure, I'll add it to my todo list :)